### PR TITLE
Small organ code clean up

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request

- Removes the `ORGAN_EXTERNAL` flag. In it's current state it has no reason to exist, `obj/item/organ/internal` should never be allowed to be marked "external" in any way.
- Refactors `/proc/getorganszone()` to be smaller and not hardcoded.
- Removes an unused proc argument.

Not big enough PR to warrant a CL.

### CODE REVIEW

File: code/modules/surgery/organs/external/tails.dm [hunk 1]
```diff
@@ -6,7 +6,6 @@
 	zone = BODY_ZONE_PRECISE_GROIN
 	slot = ORGAN_SLOT_EXTERNAL_TAIL
 	layers = EXTERNAL_FRONT|EXTERNAL_BEHIND
-	organ_flags = ORGAN_EDIBLE | ORGAN_EXTERNAL
```
Comment (Time-Green) [inline code/modules/surgery/organs/external/tails.dm:9]: stealth lizard nerf?

Comment (Kapu1178) [reply code/modules/surgery/organs/external/tails.dm:9]: Inherits ORGAN_EDIBLE from parent.

File: code/modules/surgery/organs/helpers.dm [hunk 2]
```diff
@@ -29,20 +29,14 @@
...
-			returnorg = getorganszone(BODY_ZONE_PRECISE_GROIN)
-
+/mob/living/carbon/getorganszone(zone, include_children = 0)
+	. = list()
 	for(var/obj/item/organ/organ as anything in internal_organs + external_organs)
 		if(zone == organ.zone)
-			returnorg += organ
-	return returnorg
+			. += organ
+		if(include_children && zone == deprecise_zone(organ.zone))
```
Comment (Kapu1178) [inline code/modules/surgery/organs/helpers.dm:37]:
```suggestion
		else if(include_children && zone == deprecise_zone(organ.zone))
```

Comment (Kapu1178) [reply code/modules/surgery/organs/helpers.dm]: This coulda been bad

File: code/modules/surgery/organs/external/tails.dm (hunk 1 above)
Comment (Time-Green) [reply code/modules/surgery/organs/external/tails.dm:9]: I'll allow it this one time

File: code/modules/surgery/organ_manipulation.dm [hunk 3]
```diff
@@ -205,7 +205,7 @@
 
 ///only operate on not external organs (so internal organs)
```
Comment (GoldenAlpharex) [inline code/modules/surgery/organ_manipulation.dm:206]:
```suggestion
///only operate on internal organs
```

File: code/modules/surgery/organs/helpers.dm [hunk 4]
```diff
@@ -29,20 +29,14 @@
...
-/mob/living/carbon/getorganszone(zone, subzones = 0, external = FALSE)
-	var/list/returnorg = list()
-	if(subzones)
-		// Include subzones - groin for chest, eyes and mouth for head
-		if(zone == BODY_ZONE_HEAD)
-			returnorg = getorganszone(BODY_ZONE_PRECISE_EYES) + getorganszone(BODY_ZONE_PRECISE_MOUTH)
-		if(zone == BODY_ZONE_CHEST)
-			returnorg = getorganszone(BODY_ZONE_PRECISE_GROIN)
-
+/mob/living/carbon/getorganszone(zone, include_children = 0)
```
Comment (GoldenAlpharex) [inline code/modules/surgery/organs/helpers.dm:32]:
```suggestion
/mob/living/carbon/getorganszone(zone, include_children = FALSE)
```

File: code/modules/surgery/organs/helpers.dm [hunk 5]
```diff
@@ -29,20 +29,14 @@
...
+	. = list()
 	for(var/obj/item/organ/organ as anything in internal_organs + external_organs)
 		if(zone == organ.zone)
-			returnorg += organ
-	return returnorg
+			. += organ
+		else if(include_children && zone == deprecise_zone(organ.zone))
+			. += organ
+	return .
```
Comment (GoldenAlpharex) [inline code/modules/surgery/organs/helpers.dm:39]: The usage of `.` here degrades readability and I don't think this code is hot enough to warrant a micro-optimization like that.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/helpers.dm (hunk 5 above)
Comment (JohnFulpWillard) [reply code/modules/surgery/organs/helpers.dm:39]: https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#dot-variable-

Comment (SuperNovaa41) [reply code/modules/surgery/organs/helpers.dm:39]: as in, `.` should not be being used for this

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

**Pull request merged**